src/openzeppelin/governance/timelock/library.cairo

```diff
@@ -0,0 +1,439 @@
+# SPDX-License-Identifier: MIT
+# OpenZeppelin Contracts for Cairo v0.1.0 (governance/timelock/library.cairo)
+
+%lang starknet
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.memcpy import memcpy
+from starkware.cairo.common.math import assert_le
+from starkware.cairo.common.uint256 import Uint256
+from starkware.cairo.common.bool import TRUE, FALSE
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.starknet.common.syscalls import call_contract, get_block_timestamp
+from starkware.cairo.common.math_cmp import is_le, is_not_zero
+
+from starkware.cairo.common.hash_state import (
+    HashState,
+    hash_init,
+    hash_finalize,
+    hash_update,
+    hash_update_single,
+)
+
+from openzeppelin.utils.constants import DONE_TIMESTAMP
+
+#
+# Structs
+#
+
+# tmp structs needed for external methods
+
+struct Call:
+    member to : felt
+    member selector : felt
+    member calldata_len : felt
+    member calldata : felt*
+end
+
+struct TimelockCall:
+    member to : felt
+    member selector : felt
+    member data_offset : felt
+    member data_len : felt
+end
+
+#
+# Events
+#
+
+@event
+func OperationScheduled(
```

Comment:
hm, it might be slightly cumbersome (or maybe not) but for consistency I'd try to deconstruct the Operation and emit one `CallScheduled` per call in the Multicall so this can be easily consumed next to EVM events (i.e. Solidity's)

put differently: imagine you're a frontend/backend consuming The Graph or a similar aggregator, monitoring on CairoVM and EVM timelocks, you'll want some consistency.